File: Python/Reverse-Integer/LoopSolution.py
def reverse_integer(num):
    prev_num = 0
    rev_num  = 0
    isNegative = False
    if num < 0:
        isNegative = True
        num = -num
    while num != 0:
        curr_digit = num % 10
        rev_num = rev_num*10 + curr_digit
        #overflow handler
        if ((rev_num - curr_digit) // 10) != prev_num:
            return 'Overflow, return 0 instead'
        prev_num = rev_num
        num = num // 10
    return (rev_num if not isNegative else -rev_num)
    
print(reverse_integer(123))
print(reverse_integer(-1000))
print(reverse_integer(4539))
print(reverse_integer(123134342312))

File: Java/Reverse-Integer/LoopSolution.java
public class LoopSolution{
    public static int reverse_integer(int num){
        //flag for negative input
        boolean negative = false;
        //if input is negative
        if (num<0){
            //remember
            negative = true;
            //change negative value to positive value
            num = (num*(-1));
        }
        int prevNum = 0;
        int newNum = 0;
        while (num != 0){
            //remainder
            int currentDigit = num%10;
            newNum = newNum*10 + currentDigit;
            //to handle overflow
            if((newNum-currentDigit)/10 != prevNum){
                System.out.format("Overflow, return %d instead!%n", 0);
                // throw new IllegalStateException("Overflow, return 0 instead!");
                return 0;
            }
            prevNum = newNum;
            num = num/10;
        }
        //if num is not negative return newNum, otherwise return -newNum
        return (negative==false) ? newNum : -newNum;
    }
    public static void main(String []args){
        System.out.println(reverse_integer(531));
        //leading zero output
        System.out.println(reverse_integer(100));
        //negative number
        System.out.println(reverse_integer(-321));
        //overflow
        System.out.println(reverse_integer(1000000045));
    }
}
